backend/typescript/rest/authRoutes.ts

```diff
@@ -118,4 +118,42 @@ authRouter.post(
   },
 );
 
+authRouter.post(
+  "/setPassword/:email",
+  isAuthorizedByEmail("email"),
+  async (req, res) => {
+    try {
+      const responseSuccess = await authService.setPassword(
+        req.params.email,
+        req.body.newPassword,
+      );
+      if (responseSuccess.success) {
+        const user = await userService.getUserByEmail(req.params.email);
+        if (user.status === UserStatus.INVITED) {
+          userService.updateUserById(user.id, {
+            ...user,
+            status: UserStatus.ACTIVE,
+          });
+        }
+        // automatically log in after password reset
```

Comment:
/auth/setPassword is doing 3 different things at once (setting password, updating userDTO, and logging in the user), imo the latter two actions should be handled as their own calls in the frontend (posting to "/user" and "/auth/login". That way we can handle the individual errors from each of these actions (and less code redundancy).

I think it makes more sense for this endpoint to only return the first responseSuccess from authService.setPassword (line 126).

Reply:
updated — status update cannot easily be separated from password due to authentication, but I separated login + set password.